tests/read_only_trx_overlap_timeout.py

```diff
@@ -0,0 +1,161 @@
+    specificExtraNodeosArgs[pnodes]="--read-only-threads "
+    specificExtraNodeosArgs[pnodes]+=" 2 "
+    specificExtraNodeosArgs[pnodes]+=" --max-transaction-time "
+    specificExtraNodeosArgs[pnodes]+=" 10 "
+    if args.eos_vm_oc_enable:
+        if platform.system() != "Linux":
+            Print("OC not run on Linux. Skip the test")
+            exit(True) # Do not fail the test
+        specificExtraNodeosArgs[pnodes]+=" --eos-vm-oc-enable "
+        specificExtraNodeosArgs[pnodes]+=args.eos_vm_oc_enable
+    if args.wasm_runtime:
+        specificExtraNodeosArgs[pnodes]+=" --wasm-runtime "
+        specificExtraNodeosArgs[pnodes]+=args.wasm_runtime
+
+    if cluster.launch(pnodes=pnodes, totalNodes=total_nodes, topo=topo, delay=delay, activateIF=activateIF, specificExtraNodeosArgs=specificExtraNodeosArgs, extraNodeosArgs="--resource-monitor-not-shutdown-on-threshold-exceeded" ) is False:
+        errorExit("Failed to stand up eos cluster.")
+
+    Print ("Wait for Cluster stabilization")
+    # wait for cluster to start producing blocks
+    if not cluster.waitOnClusterBlockNumSync(3):
+        errorExit("Cluster never stabilized")
+
+    producerNode = cluster.getNode()
+    apiNode = cluster.nodes[-1]
+
+
+def deployTestContracts():
+    Utils.Print("create test accounts")
+    tightloopAccount = Account(tightloopAccountName)
+    tightloopAccount.ownerPublicKey = EOSIO_ACCT_PUBLIC_DEFAULT_KEY
+    tightloopAccount.activePublicKey = EOSIO_ACCT_PUBLIC_DEFAULT_KEY
+    cluster.createAccountAndVerify(tightloopAccount, cluster.eosioAccount, buyRAM=1000000)
+
+    tightloopContractDir="unittests/test-contracts/tightloop"
+    tightloopWasmFile="tightloop.wasm"
+    tightloopAbiFile="tightloop.abi"
+    producerNode.publishContract(tightloopAccount, tightloopContractDir, tightloopWasmFile, tightloopAbiFile, waitForTransBlock=True)
+
+
+def sendTransaction(account, action, data, auth=[], opts=None):
+    trx = {
+       "actions": [{
+          "account": account,
+          "name": action,
+          "authorization": auth,
+          "data": data
+      }]
+    }
+    return apiNode.pushTransaction(trx, opts)
+
+def longROtrxThread():
+    Print("start longROtrxThread")
+
+    while stopThread==False:
+        sendTransaction(tightloopAccountName, 'doit', {"count": 50000000}, opts='--read')  #50 million is a good number to always take >10ms
+
+try:
+    startCluster()
+    deployTestContracts()
+
+    # start a background thread that constatly runs a ROtrx that is never expected to complete in max-transaction-time
+    thr = threading.Thread(target = longROtrxThread)
+    thr.start()
+
+    endTime = time.time() + testLengthSeconds
+    # and then run some other ROtrx that should complete successfully
+    while time.time() < endTime:
+        results = sendTransaction(tightloopAccountName, 'doit', {"count": 1000000}, opts='--read')  #1 million is a good number to always take <10ms
```

Comment:
Though I am a little worried about the sensitivity across runs of this time requirement

Reply:
and yup this value is no good in CI but great locally. hmmm I will have to think some on what to do. The test is more accurate the larger this number is (without "going over"). So maybe it needs to try and sniff out a good value before starting the test.

Reply:
I've pushed up some changes to the test I was hoping would get it working in CI but no... still just way too sensitive to the timing. Not sure where to go from here on it.

Reply:
I made at least one more change, cec196e, to try and improve the failures, but no luck. I think the test needs to be run with dedicated resources (really I think all the NP/LR tests should run with dedicated resources)

Reply:
One counter-point to NP/PR tests running in a more dedicated environment is that the chaos-monkey of the current approach has shaken out bugs we would never see in a more stable environment.